scripts/content.js

```diff
@@ -1,125 +1,128 @@
-let allowedMap = new Map();
-const challengeCodePrefix = '/challenge gen9nationaldexag @@@ Z-Move Clause, -Mega, Terastal Clause, Sleep Clause Mod, Forme Clause, -Hidden Power, -Last Respects, -Kings Rock, -Shadow Tag, -Acupressure, -Battle Bond, -Quick Claw, -Razor Fang, Evasion Clause, OHKO Clause, baton pass stat trap clause, -All Pokemon, +';
-
 // Polyfill for browser compatibility
-if (typeof browser === 'undefined') {
-    var browser = chrome;
-}
-
-function getChallengeCode() {
-    const pokemonList = Array.from(allowedMap.keys()).join(', +');
-    return challengeCodePrefix + pokemonList;
-}
-
-// Function to fetch JSON data
-function fetchAllowedPokemonData() {
-    browser.storage.local.get('35pokes-month').then(monthItems => {
-        browser.storage.local.get('35pokes-year').then(yearItems => {
-            browser.storage.local.get('35pokes-text').then(textItems => {
-                const xhr = new XMLHttpRequest();
-                let fileName;
-                if (monthItems['35pokes-month'] && yearItems['35pokes-year']) {
-                    textSuffix = textItems['35pokes-text'] ? '_' + textItems['35pokes-text'] : '';
-                    fileName = 'https://samuel-peter-chowdhury.github.io/35PokesShowdownFilter/dates/' + yearItems['35pokes-year'] + '_' + monthItems['35pokes-month'] + textSuffix + '.json';
-                } else {
-                    fileName = 'https://samuel-peter-chowdhury.github.io/35PokesShowdownFilter/dates/2024_5.json';
-                }
-                console.log(fileName);
-                xhr.open('GET', fileName, true);
-                xhr.onreadystatechange = function() {
-                    if (xhr.readyState == 4) {
-                        if (xhr.status === 200) {
-                            const data = JSON.parse(xhr.responseText);
-                            data.forEach(item => {
-                                allowedMap.set(item.toLowerCase(), true);
-                            });
-                            browser.storage.local.set({ '35pokes-code': getChallengeCode() }, function() {});
-                            console.log(allowedMap);
-                        } else {
-                            alert('Invalid Date');
-                        }
-                    }
-                }
-                xhr.send();
-            });
-        });
-    });
-}
-
-// Call the function to fetch JSON data
-fetchAllowedPokemonData();
+if (typeof browser === "undefined") globalThis.browser = chrome;

```

Comment:
Not specific to this line of code. But this refactor does not work on my end. I am using the latest version of chrome on a windows pc. When the date is changed, it is not reflected immediately in the teambuilder, it still requires a reload. Additionally, it seems to have introduced a new bug that wipes all pokemon from the list in certain situations.

Reply:
The changes work as expected on my end in the same environment. However, the behavior you describe sounds as if this edit in manifest was omitted: https://github.com/swordfishtr/35PokesShowdownFilter/blob/314d32eee4fb0d412e820a84973a8ec902fca433/manifest.json#L25

The new bug was introduced into the showdown client a while ago - it happens if you view the Pokemon pool before the resources for the selected format are loaded. Naturally, only replicable if a format that has resources is selected (like gen9ou). And the resources stay loaded until tab reload.